packages/angular_devkit/schematics/src/workflow/base.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { logging, schema, virtualFs } from '@angular-devkit/core';
+import { Observable, Subject, concat, of, throwError } from 'rxjs';
+import { concatMap, defaultIfEmpty, ignoreElements, last, map, tap } from 'rxjs/operators';
+import { EngineHost, SchematicEngine } from '../engine';
+import { UnsuccessfulWorkflowExecution } from '../exception/exception';
+import { standardFormats } from '../formats';
+import { DryRunEvent, DryRunSink } from '../sink/dryrun';
+import { HostSink } from '../sink/host';
+import { HostTree } from '../tree/host-tree';
+import { Tree } from '../tree/interface';
+import { optimize } from '../tree/static';
```

Comment:
Not something that should be addressed in this PR.  But is there something we can do to limit the volume of independent imports?  This seems like it could lead to a refactoring nightmare in the future.

Reply:
Without having circular dependencies, not sure.